programs/solana/programs/ics26-router/src/instructions/client.rs

```diff
@@ -135,3 +135,333 @@ pub struct ClientStatusUpdatedEvent {
     pub client_id: String,
     pub active: bool,
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::state::RouterState;
+    use anchor_lang::{AnchorDeserialize, InstructionData};
+    use mollusk_svm::result::Check;
+    use mollusk_svm::Mollusk;
+    use solana_sdk::account::Account;
+    use solana_sdk::instruction::{AccountMeta, Instruction};
+    use solana_sdk::pubkey::Pubkey;
+    use solana_sdk::{native_loader, system_program};
+
+    fn create_account_data(
+        account_name: &str,
+        init_space: usize,
+        serialize_fn: impl FnOnce(&mut [u8]),
+    ) -> Vec<u8> {
+        let mut data = vec![0u8; 8 + init_space];
+
+        // Write discriminator
+        let discriminator: [u8; 8] =
+            anchor_lang::solana_program::hash::hash(format!("account:{account_name}").as_bytes())
+                .to_bytes()[..8]
+                .try_into()
+                .unwrap();
+        data[0..8].copy_from_slice(&discriminator);
+
+        // Serialize account data
+        serialize_fn(&mut data[8..]);
+
+        data
+    }
+
+    fn serialize_string(data: &mut [u8], offset: &mut usize, value: &str) {
+        let bytes = value.as_bytes();
+        let len = bytes.len() as u32;
+        data[*offset..*offset + 4].copy_from_slice(&len.to_le_bytes());
+        *offset += 4;
+        data[*offset..*offset + bytes.len()].copy_from_slice(bytes);
+        *offset += bytes.len();
+    }
+
+    fn serialize_vec_u8(data: &mut [u8], offset: &mut usize, value: &[u8]) {
+        let len = value.len() as u32;
+        data[*offset..*offset + 4].copy_from_slice(&len.to_le_bytes());
+        *offset += 4;
+        data[*offset..*offset + value.len()].copy_from_slice(value);
+        *offset += value.len();
+    }
```

Comment:
These functions are very hard to understand/read. I'm ok with it if we absolutely need them